site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,751 @@
 
-en.yaml
+Here is an example translation file for English, called `locales/en.ftl`:
 
-```yaml
-hey: Hello!
-help:
-  short: This bot can help you with stuff.
-  long: >
-    This bot can help you with stuff.
-    For example…
+```fluent
+start = Hi, how can I /help you?
+help = 
+    Send me some text, and I can make it bold for you.
+    You can change my language using the /language command.
 ```
 
-de.yaml
+The German equivalent would be called `locales/de.ftl` and look like this:
 
-```yaml
-hey: Hallo!
-help:
-  short: Dieser Bot kann dir bei Dingen helfen.
-  long: >
-    Dieser Bot kann dir bei Dingen helfen.
-    Zum Beispiel…
+```fluent
+start = Hallo, wie kann ich dir helfen? /help
+help = 
+    Schick eine Textnachricht, die ich für dich fett schreiben soll.
+    Du kannst mit dem Befehl /language die Spache ändern.
 ```
 
-As you can see the same keys are existing in both files.
-When we now want to send a message to a user with have to use the key.
+In your bot, you can now use these translations through the plugin.
+It will make them available through `ctx.t`:
 
 ```ts
 bot.command("start", async (ctx) => {
-  await ctx.reply(ctx.i18n.t("hey"));
+  await ctx.reply(ctx.t("start"));
+});
+
+bot.command("help", async (ctx) => {
+  await ctx.reply(ctx.t("help"));
+});
+```
+
+Whenever you call `ctx.t`, the locale of the current context object `ctx` is used to find the proper translation.
+Finding the proper translation is done using a function called _localeNegotiator_.
+In the simplest case, it just returns `ctx.from.language_code`.
+
+As a result, users with different locales will be able to read the messages, each in their language.
+
+## Usage
+
+The plugin derives the user's locale from many different factors.
+One of them is from `ctx.from.language_code`, which will be provided by the user's client.
+
+However, there are many more things that can be used to determine the user's locale.
+For example, you could store the user's locale in your [session](./session.md).
+Hence, there are two main ways to use this plugin: [With Sessions](#with-sessions) and [Without Sessions](#without-sessions).
+
+### Without Sessions
+
+It's easier to use and set up the plugin without sessions.
+Its main drawback is that you can't store the languages the users choose.
+
+Like mentioned above, the locale to be used for the user will be decided with `ctx.from.language_code`, which is coming from the user's client.
+But the default language will be used if you don't have a translation of that language.
+Sometimes your bot might not be able to see the user's preferred language provided by their client, and in that case the default language will be used, too.
+The `ctx.from.language_code` will be visible only if the user previously has started a private conversation with your bot.
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+import { Bot, Context } from "grammy";
+import { I18n, I18nFlavor } from "@grammyjs/i18n";
```

Comment:
Should we encourage users to do this? Is this good practice? Is this irrelevant and confusing?
```suggestion
import { Bot, type Context } from "grammy";
import { I18n, type I18nFlavor } from "@grammyjs/i18n";
```
If you apply this suggestion, please adjust the remaining code snippets, too.

Reply:
For me, its not irrelevant and confusing. But I didn't include them because, I was focusing on simple examples - minimal as possible. And, I didn't find other pages having explicit type imports (example: session, router, etc). So, to be consistent with the rest of the documentation I think we shouldn't be doing this here.